promotion.md

```diff
@@ -111,6 +121,41 @@ it is unlikely to be the desired behavior in most cases and very likey to be con
 to the user. If such behavior is desired, the user can still use an explicit `static`
 or `const` item and refer to that.
 
+*Dynamic check.* The Miri engine could dynamically check this by ensuring that
+the result of computing a promoted is a value that does not need dropping.
+
+## `&` in `const` and `static`
+
+Promotion is also responsible for making code like this work:
+
+```rust
+const FOO: &'static i32 = {
+    let x = &13;
+    x
+};
+```
+
+However, since this is in explicit const context, we are less strict about
+promotion in this situation: all function calls are promoted, not just
+`#[rustc_promotable]` functions.
```

Comment:
@RalfJung  I think my confusion really occurs because of this paragraph. It seems to apply that *both* `const`s should compile in the following example, which is not true.

Reply:
Ah yes. That's because of the `Drop` check, which still applies.

Reply:
It also fails on types [which are not `Drop`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=30be96601b71d20a232d99e610e394d4). I think @oli-obk comment above was incorrect?

Reply:
That example [works](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=92003e9d833ac8dc9f6b63eb6720aa8b) if you make `new` a `const fn`.